File: backend/src/main/java/com/davnig/invoicemanagementapi/service/TaxableSubjectService.java
package com.davnig.invoicemanagementapi.service;

import com.davnig.invoicemanagementapi.model.dto.Paginating;
import com.davnig.invoicemanagementapi.model.dto.TaxableSubjectSummary;
import com.davnig.invoicemanagementapi.model.entity.TaxableSubject;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TaxableSubjectService {

    Page<TaxableSubjectSummary> findAll(Paginating paginating, String search, List<String> fields);

    Page<TaxableSubjectSummary> findAll(Paginating paginating, String search);

    Page<TaxableSubjectSummary> findAll(Paginating paginating, List<String> fields);

    Page<TaxableSubjectSummary> findAll(Paginating paginating);

    TaxableSubject findById(int id);

    TaxableSubject findById(int id, List<String> fields);

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/service/TaxableSubjectService.kt
package com.davnig.invoicemanagementapi.service

import com.davnig.invoicemanagementapi.model.dto.Paginating
import com.davnig.invoicemanagementapi.model.dto.TaxableSubjectDetail
import com.davnig.invoicemanagementapi.model.dto.TaxableSubjectSummary
import org.springframework.data.domain.Page

interface TaxableSubjectService {

    fun findAll(paginating: Paginating, search: String, fields: List<String>): Page<TaxableSubjectSummary>

    fun findAll(paginating: Paginating, search: String): Page<TaxableSubjectSummary>

    fun findAll(paginating: Paginating, fields: List<String>): Page<TaxableSubjectSummary>

    fun findAll(paginating: Paginating): Page<TaxableSubjectSummary>

    fun findById(id: Int): TaxableSubjectDetail

    fun findById(id: Int, fields: List<String>): TaxableSubjectDetail

}